cms/static/js/views/modals/course_outline_modals.js

```diff
@@ -591,20 +580,69 @@ define(['jquery', 'backbone', 'underscore', 'gettext', 'js/views/baseview',
             return this.model.get('ancestor_has_staff_lock');
         },
 
+        getContentGroups: function() {
+            return this.model.get('user_partitions');
+        },
+
+        getGroupAccess: function() {
+            return this.model.get('group_access');
+        },
+
         getContext: function() {
             return {
                 hasExplicitStaffLock: this.isModelLocked(),
-                ancestorLocked: this.isAncestorLocked()
+                ancestorLocked: this.isAncestorLocked(),
+                contentGroups: this.getContentGroups(),
+                groupAccess: this.getGroupAccess()
             };
-        }
+        },
     });
 
     StaffLockEditor = AbstractVisibilityEditor.extend({
         templateName: 'staff-lock-editor',
         className: 'edit-staff-lock',
+        events: {
+            'change .user-partition-select': function() {
+                this.hideCheckboxDivs();
+                this.showSelectedDiv(this.getSelectedEnrollmentTrackId());
+                this.getSelectedGroupAccess();
+            }
+        },
+
         afterRender: function() {
             AbstractVisibilityEditor.prototype.afterRender.call(this);
             this.setLock(this.isModelLocked());
+            this.hideCheckboxDivs();
+            for (var key in this.model.attributes.group_access) {
+                this.$('.user-partition-select').val(key).change(); // should be only one partition key
+            }
+        },
+
+        getSelectedEnrollmentTrackId: function() {
```

Comment:
Can you guys pull in VisibilityEditorView from /edx-platform/cms/static/js/xblock/authoring.js (or refactor it so that the guts can be shared between these 2 JS classes?